packages/pluggableWidgets/background-gradient-native/src/BackgroundGradient.tsx

```diff
@@ -0,0 +1,45 @@
+import { ReactElement, createElement } from "react";
+import { Pressable, StyleSheet, NativeModules, Alert } from "react-native";
+import LinearGradient from "react-native-linear-gradient";
+
+import { executeAction } from "@mendix/piw-utils-internal";
+import { mergeNativeStyles } from "@mendix/pluggable-widgets-tools";
+
+import defaultStyle, { CustomStyle } from "./ui/Styles";
+
+import { BackgroundGradientProps } from "../typings/BackgroundGradientProps";
+
+export type props = BackgroundGradientProps<CustomStyle>;
+
+export function BackgroundGradient({ name, colorList, angle, content, opacity, onClick, style }: props): ReactElement {
+    if (!("BVLinearGradient" in NativeModules.UIManager)) {
+        Alert.alert("", "The widget 'Background gradient' requires an updated 'Make It Native 9' application");
+    }
+
+    let sortedColorList = colorList.sort((a, b) => Number(a.offset) - Number(b.offset));
+    // checking if the color list only have one color item , then we should duplicate it. one color list throws an exception on android.
+    if (sortedColorList.length === 1) {
+        sortedColorList = [...sortedColorList, ...sortedColorList];
+    }
+    const colors = sortedColorList.map(colorsObject => colorsObject.color.toLowerCase());
+    const offsets = sortedColorList.map(colorsObject => Number(colorsObject.offset));
+    const styles = mergeNativeStyles(defaultStyle, style);
+    return (
+        <Pressable
+            onPress={() => {
+                executeAction(onClick);
+            }}
+            testID={`background-gradient-${name}`}
```

Comment:
@Ahmad-Elsayed We should not append the widget name. This differs from our current standards of defining test id. Please refer test id in other native widgets for instance. Our MxDevs would not come to know what is the prefix value whenever they want to write the e2e tests for their own project. 

@UrazAkgultan FYI, please make a note of it.